mongocrypt/src/binary.rs

```diff
@@ -0,0 +1,74 @@
+use mongocrypt_sys as sys;
+
+use crate::convert::binary_bytes;
+use crate::error::Result;
+
+pub(crate) struct Binary {
+    inner: *mut sys::mongocrypt_binary_t,
+}
+
+impl Drop for Binary {
+    fn drop(&mut self) {
+        unsafe {
+            sys::mongocrypt_binary_destroy(self.inner);
+        }
+    }
+}
+
+impl Binary {
+    pub(crate) fn new() -> Self {
+        Self {
+            inner: unsafe { sys::mongocrypt_binary_new() },
+        }
+    }
+
+    pub(crate) fn native(&self) -> *mut sys::mongocrypt_binary_t {
```

Comment:
I'm not sure if this could be a problem in Rust too, but in the Swift driver we ran into some subtle issues directly exposing pointers to C types from their wrapping Swift types. things like this could happen:
```swift
let obj = SomeType() // backed by a pointer to a C type
let ptr = obj.backing_ptr 
// `obj` gets deinitialized/dropped here, which calls the `destroy` method on the backing C type
doSomething(ptr) // this fails, because ptr is no longer valid
```

the standard safe way to handle those cases in Swift is to have methods that accept closures that take in pointers, something like
```swift
obj.withBackingPtr({ ptr in 
    doSomething(ptr)
})
```
that way, we know `obj` definitely will not be dropped yet, since we only use the pointer within a method on `obj`.

should we be concerned about anything like that with these `native` methods?